src/components/Tabs/Tabs.tsx

```diff
@@ -1 +1,59 @@
-export const Tabs = () => {};
+/* eslint-disable no-console */
+import { useEffect } from 'react';
+import { Tab } from '../../types/tab';
+
+type Props = {
+  tabs: Array<Tab>,
+  selectedTab: Tab,
+  onTabSelected: (tab: Tab) => void,
+};
+
+export const Tabs = ({
+  tabs, selectedTab, onTabSelected,
+}: Props) => {
+  useEffect(() => {
+    if (!tabs.some(t => t === selectedTab)) {
+      onTabSelected(tabs[0]);
+    }
+  });
+
+  const onClickHandler = (event: React.MouseEvent, tab: Tab) => {
+    event.preventDefault();
+    if (tab !== selectedTab) {
+      onTabSelected(tab);
+    }
```

Comment:
JS isn't capable of comparing objects, it can compare only primitive values, please try to compare ids instead of objects